### PR TITLE
some gnmi.Get() call is replaced by gnmi.Watch() call

### PR DESCRIPTION
**Pull request by abhijit-dhar**
The test "feature/gribi/otg_tests/get_rpc_test/get_rpc_test.go" was failing against CISCO vxr dut (kne environment). The test was not getting the expected results by placing gnmi.Get() queries into the DUT. So, I replaced the "gnmi.Get()" query with "gnmi.Watch()/gnmi.Get()". Check the value is present by polling the DUT and if the value is found then fetch and compare.

### CODE REVIEW

File: feature/gribi/otg_tests/get_rpc_test/get_rpc_test.go [hunk 1]
```diff
@@ -317,8 +318,18 @@ func testIPv4LeaderActive(ctx context.Context, t *testing.T, args *testArgs) {
...
 	for ip := range ateDstNetCIDR {
 		ipv4Path := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(args.dut)).Afts().Ipv4Entry(ateDstNetCIDR[ip])
-		if got, want := gnmi.Get(t, args.dut, ipv4Path.State()).GetPrefix(), ateDstNetCIDR[ip]; got != want {
-			t.Errorf("ipv4-entry/state/prefix got %s, want %s", got, want)
+		lastValue, ok := gnmi.Watch(t, args.dut, ipv4Path.State(), time.Minute, func(v *ygnmi.Value[*oc.NetworkInstance_Afts_Ipv4Entry]) bool {
+			return v.IsPresent()
+		}).Await(t)
+
+		if ok {
+			if got, want := gnmi.Get(t, args.dut, ipv4Path.State()).GetPrefix(), ateDstNetCIDR[ip]; got != want {
```
Comment (greg-dennis) [inline feature/gribi/otg_tests/get_rpc_test/get_rpc_test.go:326]: Replace "gnmi.Get(t, args.dut, ipv4Path.State())" with lastValue

File: feature/gribi/otg_tests/get_rpc_test/get_rpc_test.go [hunk 2]
```diff
@@ -345,10 +356,20 @@ func testIPv4LeaderActive(ctx context.Context, t *testing.T, args *testArgs) {
...
-		if got, want := gnmi.Get(t, args.dut, ipv4Path.State()).GetPrefix(), ateDstNetCIDR[ip]; got != want {
-			t.Errorf("ipv4-entry/state/prefix got %s, want %s", got, want)
-		}
+	        ipv4Path := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(args.dut)).Afts().Ipv4Entry(ateDstNetCIDR[ip])
+                lastValue, ok := gnmi.Watch(t, args.dut, ipv4Path.State(), time.Minute, func(v *ygnmi.Value[*oc.NetworkInstance_Afts_Ipv4Entry]) bool {
+                        return v.IsPresent()
+                }).Await(t)
+
+                if ok {
+                        if got, want := gnmi.Get(t, args.dut, ipv4Path.State()).GetPrefix(), ateDstNetCIDR[ip]; got != want {
```
Comment (greg-dennis) [inline feature/gribi/otg_tests/get_rpc_test/get_rpc_test.go:365]: replace "gnmi.Get(t, args.dut, ipv4Path.State())" with lastValue

File: feature/gribi/otg_tests/get_rpc_test/get_rpc_test.go [hunk 3]
```diff
@@ -317,8 +318,18 @@ func testIPv4LeaderActive(ctx context.Context, t *testing.T, args *testArgs) {
 	// Verify the above entries are active through AFT Telemetry.
 	for ip := range ateDstNetCIDR {
 		ipv4Path := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(args.dut)).Afts().Ipv4Entry(ateDstNetCIDR[ip])
-		if got, want := gnmi.Get(t, args.dut, ipv4Path.State()).GetPrefix(), ateDstNetCIDR[ip]; got != want {
-			t.Errorf("ipv4-entry/state/prefix got %s, want %s", got, want)
+		lastValue, ok := gnmi.Watch(t, args.dut, ipv4Path.State(), time.Minute, func(v *ygnmi.Value[*oc.NetworkInstance_Afts_Ipv4Entry]) bool {
+			return v.IsPresent()
+		}).Await(t)
+
+		if ok {
```
Comment (greg-dennis) [inline feature/gribi/otg_tests/get_rpc_test/get_rpc_test.go:325]: more idiomatic to check the error condition ("if !ok") first and then the non-error case doesn't need to be in "else"

File: feature/gribi/otg_tests/get_rpc_test/get_rpc_test.go [hunk 4]
```diff
@@ -345,10 +356,20 @@ func testIPv4LeaderActive(ctx context.Context, t *testing.T, args *testArgs) {
...
-		ipv4Path := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(args.dut)).Afts().Ipv4Entry(ateDstNetCIDR[ip])
-		if got, want := gnmi.Get(t, args.dut, ipv4Path.State()).GetPrefix(), ateDstNetCIDR[ip]; got != want {
-			t.Errorf("ipv4-entry/state/prefix got %s, want %s", got, want)
-		}
+	        ipv4Path := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(args.dut)).Afts().Ipv4Entry(ateDstNetCIDR[ip])
+                lastValue, ok := gnmi.Watch(t, args.dut, ipv4Path.State(), time.Minute, func(v *ygnmi.Value[*oc.NetworkInstance_Afts_Ipv4Entry]) bool {
+                        return v.IsPresent()
+                }).Await(t)
+
+                if ok {
```
Comment (greg-dennis) [inline feature/gribi/otg_tests/get_rpc_test/get_rpc_test.go:364]: more idiomatic to check the error condition ("if !ok") first and then the non-error case doesn't need to be in "else"

**Review by greg-dennis — CHANGES_REQUESTED** [review]

File: feature/gribi/otg_tests/get_rpc_test/get_rpc_test.go (hunk 1 above)
Comment (abhijit-dhar) [reply feature/gribi/otg_tests/get_rpc_test/get_rpc_test.go]: Agreed and implemented.

**Review by self-maurya — APPROVED** [review]

**Review by dplore — APPROVED** [review]

**Pull request merged**